docs/integration/elasticsearch-integration.md

```diff
@@ -52,6 +52,136 @@ Last, but certainly not least, you may want to get an alert if a node leaves the
+
+#### Actions to take
+
+- Check the status of nodes within the Elasticsearch cluster for any nodes that may be offline, unavailable, or experiencing issues
+- If node failures are detected, you may need to restart failed nodes or replace hardware
+- If the node count changes due to scaling activities (e.g., adding or removing nodes), review the recent scaling events to confirm that they are intentional and expected
+- Monitor network connectivity between nodes within the Elasticsearch cluster for any network issues that may be affecting communication and node discovery
+
+
+### Java old gen usage > 97%
+
+This alert rule continuously monitors the usage of Java's old generation heap memory in an Elasticsearch environment, triggering a warning if the usage exceeds 97%. The minimum delay between consecutive alerts triggered by this alert rule is set to 10 minutes.
+
+Suppose the Java old generation heap memory usage in the Elasticsearch environment starts increasing and eventually exceeds 97% over a 5-minute period.
+
+#### Actions to take
+
+- Analyze the application's memory usage patterns for any memory leaks
+- Review and optimize the Java Virtual Machine (JVM) configuration, including heap size settings, garbage collection algorithms, and memory management parameters
+- Monitor system resources, including memory, CPU, and disk I/O, and consider scaling them up if necessary
+- Investigate recent application changes, updates, or deployments that may have contributed to the spike in memory usage
+
+### Field data size
+
+This alert rule continuously monitors the field data size in an Elasticsearch cluster and triggers a warning if the field data size exceeds a certain threshold (20 in this case). The minimum delay between consecutive alerts triggered by this alert rule is set to 10 minutes.
+
+Suppose the field data size in the Elasticsearch cluster increases significantly due to increased query loads or inefficient queries. When this happens, the alert rule checks the field data size over the last 10 minutes. If it exceeds 20% of the JVM heap committed memory during this timeframe, the alert rule triggers a warning.
+
+#### Actions to take
+
+- Analyze the query patterns and usage that contribute to the increased field data size. Look for inefficient or resource-intensive queries that may need optimization
+- Optimize queries to reduce the amount of field data loaded into memory. This may involve restructuring queries, using filters, or optimizing aggregations to minimize memory usage
+- Consider scaling up the resources allocated to the Elasticsearch cluster, such as increasing the JVM heap size, to accommodate the increased field data size
+
+
+### Tripped parent circuit breaker
```

Comment:
Parent Breaker has two modes:
- the default, on "real data". Meaning it looks at the current heap usage before doing an operation. If heap usage > threshold --> operation is not run, because you're in danger of running out of heap
- real_data=false --> just the sum of all the other breakers. I.e. if an operation tries to load memory that would use more than X% of heap across all breakers (e.g. field data, in-flight requests...), that operation isn't run

Reply:
Thanks Radu. So we alert on any Parent circuit breaker exception. If real memory is used it will be triggered at 95% of JVM heap, otherwise at 70% due to estimates. I guess both are configurable (?). Do you think we need to clarify anything here? Just mention the settings maybe?

Reply:
Right, 95% is the current default for real memory, 70% for "sum-of-all-circuit-breakers". I think we can safely assume people have real memory enabled, I didn't really see clusters without real memory. And yes, it's configurable.

As for clarifications, I think it's important to say that the parent CB triggered implies that the memory usage right now is very high.